### PR TITLE
 JAMES-2550 Cassandra MailQueueView support for deletion 

### PR DESCRIPTION
**Pull request by aduprat**
Only 3 last commits.

### CODE REVIEW

File: core/src/main/java/org/apache/james/core/MailAddress.java [hunk 1]
```diff
@@ -665,4 +665,14 @@ private int parseDomain(StringBuffer dSB, String address, int pos) throws Addres
...
         return false;
     }
+
+
+    /**
+     * Return <code>true</code> if the {@link MailAddress} equals the given address ignoring the case.
+     *
+     * @return nullsender
+     */
+    public boolean equalsIgnoreCase(String address) {
```
Comment (blackheaven) [inline core/src/main/java/org/apache/james/core/MailAddress.java]: We assume that two usernames can't be distinguished from case sensitivity ?
I can't find this rule in the RFC.

Comment (aduprat) [reply core/src/main/java/org/apache/james/core/MailAddress.java]: In this RFC: https://tools.ietf.org/html/rfc5321#section-2.4
```
...
Mailbox domains follow normal DNS rules and are
   hence not case sensitive.
...
```

Comment (blackheaven) [reply core/src/main/java/org/apache/james/core/MailAddress.java]: I agree for domains, but for user/username? what is its status?

Comment (mbaechler) [reply core/src/main/java/org/apache/james/core/MailAddress.java]: the user part of a email address doesn't have rules. It's up to the authoritative server for this domain to decide how this part is handled.

Comment (mbaechler) [reply core/src/main/java/org/apache/james/core/MailAddress.java]: hence the "don't try to validate email address" legendary claim

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/view/api/DeleteConditionTest.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+            assertThat(
+                DeleteCondition.withName(NAME)
+                    .shouldBeDeleted(FakeMail.builder()
+                        .name(NAME)
+                        .build()))
+                .isTrue();
+        }
+
+        @Test
+        void withSenderShouldReturnFalseWhenDifferentAddress() throws Exception {
```
Comment (rouazana) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/view/api/DeleteConditionTest.java]: s/withSender/withName/ ?

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/view/api/DeleteConditionTest.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+            assertThat(
+                DeleteCondition.withRecipient(ADDRESS)
+                    .shouldBeDeleted(FakeMail.builder()
+                        .recipient(ADDRESS)
+                        .build()))
+                .isTrue();
+        }
+
+        @Test
+        void withRecipientShouldReturnTrueWhenAtListOneMatches() throws Exception {
```
Comment (rouazana) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/view/api/DeleteConditionTest.java]: s/List/Least

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/view/api/DeleteConditionTest.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        }
+
+        @Test
+        void recipientShouldBeConvertedToWithRecipient() {
+            assertThat(DeleteCondition.from(ManageableMailQueue.Type.Recipient, VALUE))
+                .isEqualTo(DeleteCondition.withRecipient(VALUE));
+        }
+
+        @Test
+        void nameShouldBeConvertedToWithBName() {
```
Comment (rouazana) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/view/api/DeleteConditionTest.java]: s/BName/Name/

File: core/src/main/java/org/apache/james/core/MailAddress.java [hunk 5]
```diff
@@ -665,4 +665,14 @@ private int parseDomain(StringBuffer dSB, String address, int pos) throws Addres
     public boolean isNullSender() {
         return false;
     }
+
+
+    /**
+     * Return <code>true</code> if the {@link MailAddress} equals the given address ignoring the case.
+     *
+     * @return nullsender
```
Comment (rouazana) [inline core/src/main/java/org/apache/james/core/MailAddress.java]: wrong ^ ^

File: core/src/main/java/org/apache/james/core/MailAddress.java (hunk 1 above)
Comment (mbaechler) [reply core/src/main/java/org/apache/james/core/MailAddress.java]: No, it doesn't work like that : you can't decide what are the comparison rules for the authoritative server of that domain, you can only decide for mail address you own.

Comment (chibenwa) [reply core/src/main/java/org/apache/james/core/MailAddress.java]: IMO we need to
 - Compare case sensitively the username
 - Compare in a case sensitive fashion the domain name

Case insensitiveness of domains is already e,beded as part of `Domain` object which MailAddress relies on.

Tests needs to be added in MailAddress tests:

```
a@d.com == a@D.com
A@d.com != a@d.com
```

But this commit should be dropped.

(BTW equals already do ignore case assertions...)

```
    @Override
    public final boolean equals(Object obj) {
        if (obj == null) {
            return false;
        } else if (obj instanceof String) {
            String theString = (String) obj;
            return toString().equalsIgnoreCase(theString);
        } else if (obj instanceof MailAddress) {
            MailAddress addr = (MailAddress) obj;
            if (isNullSender() && addr.isNullSender()) {
                return true;
            }
            return equalsIgnoreCase(getLocalPart(), addr.getLocalPart())
                && Objects.equals(getDomain(), addr.getDomain());
        }
        return false;
    }
```

So I wonder how `equalsIgnoreCase` differs from `equals`. We may need a (separate) ticket to fix this incorrect behaviour (James should support case variation in mailAdresss - or alike)

Maybe TDD would have avoided such duplicated code BTW...

**Comment by aduprat** [comment]
#1754

**Pull request closed without merging**